woocommerce/woocommerce-ios
# A quick example using the badgeValue property

Fixes #1576
Fixes #1601 

Since this proposal has dev & design support, I have converted the draft PR to a normal PR.

~This PR is meant to demo the order badge using the `badgeValue` property found in `UITabBarItem`. It's not merge-worthy right now because it doesn't remove the manual orderBadge creation code. If this proposal is approved, I will change this to a real PR and make sure the custom code is removed.~

(Kylea let me know if you need any other screenshot types.)

### Changes
- removes any references to the `OrdersBadgeController`
- replaces them with the `badgeValue` property in `UITabBarItem`
- deletes the custom control, `OrdersBadgeController`

### To test
- run the app on any device
- transition from portrait to landscape and back again to check the orientation of the badge
- use a RTL language, activate Products, and ensure the badge remains on the Orders tab (instead of Products)
- transition from light mode to dark mode
- try to break it in general

---

Original testing instructions

- [x] Does is work for all screen sizes? Previously we had issues with the iPad's badge position and iPhone 8's badge position, when using the custom badge code.
- [x] Does it transition colors correctly between dark mode and light mode?
- [x] Does it pass design standards? Note that the `badgeValue` property does not allow a border and it does not allow us to control the positioning.

**iPhone 11 Pro Max**
<img src="https://user-images.githubusercontent.com/1062444/70554486-d09d9080-1b42-11ea-8f8b-23c6329f8435.png" width="400" />

![Simulator Screen Shot - iPhone 11 Pro Max - 2019-12-10 at 11 34 19](https://user-images.githubusercontent.com/1062444/70554605-080c3d00-1b43-11ea-852f-cf257062a49e.png)

**iPhone 8**
Select an image to see a larger version.

| Light Mode | Dark Mode | 
| --- | --- |
| <img src="https://user-images.githubusercontent.com/1062444/70554947-bdd78b80-1b43-11ea-9e2b-267635730713.png" width="350" /> | <img src="https://user-images.githubusercontent.com/1062444/70555034-df387780-1b43-11ea-97a5-5a71b2be643c.png" width="350" /> |

<kbd>![Simulator Screen Shot - iPhone 8 - 2019-12-10 at 11 52 48](https://user-images.githubusercontent.com/1062444/70555017-d8116980-1b43-11ea-9886-3f0e362ef285.png)</kbd>

<kbd>![Simulator Screen Shot - iPhone 8 - 2019-12-10 at 11 53 29](https://user-images.githubusercontent.com/1062444/70555166-232b7c80-1b44-11ea-9208-12c8309883ec.png)</kbd>

**iPad Pro (10.5 inch)**
Select an image to see a larger version.

<kbd><img src="https://user-images.githubusercontent.com/1062444/70555443-a220b500-1b44-11ea-9b46-754117e4a81a.png" width="400" /></kbd>

<kbd>![Simulator Screen Shot - iPad Pro (10 5-inch) - 2019-12-10 at 11 59 22](https://user-images.githubusercontent.com/1062444/70555450-a77dff80-1b44-11ea-985b-17c1f03679f9.png)</kbd>

**iPad Pro (10.5 inch) with 9+ orders**
<kbd><img src="https://user-images.githubusercontent.com/1062444/70555895-88cc3880-1b45-11ea-8298-eb774b00002a.png" width="400" /></kbd>

<kbd>![Simulator Screen Shot - iPad Pro (10 5-inch) - 2019-12-10 at 12 06 18](https://user-images.githubusercontent.com/1062444/70555912-95e92780-1b45-11ea-829f-1629ec6f70bc.png)</kbd>

**iPhone Xs**
<kbd><img src="https://user-images.githubusercontent.com/1062444/70555563-e6ac5080-1b44-11ea-9d38-a53e2bbcbc4f.png" width="350" /></kbd>

<kbd>![Simulator Screen Shot - iPhone Xs - 2019-12-10 at 12 01 47](https://user-images.githubusercontent.com/1062444/70555595-f6c43000-1b44-11ea-8f96-7fdee10210aa.png)</kbd>

**9+ Orders on iPhone Xs**
<kbd><img src="https://user-images.githubusercontent.com/1062444/70555691-2d9a4600-1b45-11ea-9425-2195a4e40993.png" width="350" /></kbd>

<kbd>![Simulator Screen Shot - iPhone Xs - 2019-12-10 at 12 03 33](https://user-images.githubusercontent.com/1062444/70555694-2ffca000-1b45-11ea-85ae-978cdb640753.png)</kbd>


Update release notes:
